### PR TITLE
sort added in user and classroom

### PR DESCRIPTION
**Pull request by ashish0910**
issue #50 fixed
Sort by name

Sort by created time

### CODE REVIEW

File: dashboard/controller/classrooms.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ exports.index = function(req, res) {
 
 	//query
 	var query = {
-		sort: '+name'
+	sort: '-name'
```
Comment (tarunsinghal92) [inline dashboard/controller/classrooms.js]: indentation issue

**Comment by ashish0910** [comment]
sorting test for user 

sort by name 

name

sort by created time

created time

sort by last seen

last seen

Testing for classroom already added .
@tarunsinghal92 Changes done please check .

**Comment by ashish0910** [comment]
@tarunsinghal92 here is result of `npm test`

ashishs-MacBook-Air:sugarizer-server ashishaggarwal$ npm test

> sugarizer-server@1.1.0-alpha test /Users/ashishaggarwal/Desktop/stuff/sugar/server/sugarizer-server
> mocha api/test/

   _____                        _
  / ____|                      (_)
 | (___  _   _  __ _  __ _ _ __ _ _______ _ __
  \___ \| | | |/ _` |/ _` | '__| |_  / _ \ '__|
  ____) | |_| | (_| | (_| | |  | |/ /  __/ |
 |_____/ \__,_|\__, |\__,_|_|  |_/___\___|_|
                __/ |
               |___/
Server for Sugarizer application v1.1.0-alpha
Settings file './env/test.ini'


Presence is listening on port 8039...
API is listening on port 8080...
  Activities
    /GET activities
      ✓ it should return all the activities
      ✓ it should return all fields (50ms)
      ✓ it should return right number of favorites
      ✓ it should return all activities in descending order of "name" with fields [id,name]
      ✓ it should return right activity with name "Paint" and fields [id,name]
    /GET/:id activities
      ✓ it should return nothing on inexisting activity
      ✓ it should return right activity on existing id
      ✓ it should return right activity with fields [id,name]
    /PUT activities
      ✓ it should update favorites and order property of activities
(node:1588) DeprecationWarning: collection.remove is deprecated. Use deleteOne, deleteMany, or bulkWrite instead.

  Classrooms
    /POST classroom
      ✓ it should add a classroom
    /GET/:id classroom
      ✓ it should return nothing on invalid id
      ✓ it should return nothing on inexisting id
      ✓ it should return right classroom by id
    /GET classrooms
      ✓ it should return all the classrooms
      ✓ it should return all the fields for classrooms
      ✓ it should return all matched classrooms with partial search
    /PUT/:id classrooms
      ✓ it should do nothing on invalid classroom
      ✓ it should do nothing on inexisting classroom
      ✓ it should update the valid classroom
    /DELETE/:id classrooms
      ✓ it should do nothing on invalid classroom
      ✓ it should not remove an inexisting classroom
      ✓ it should remove the valid classroom

  Journal
    /GET shared
      ✓ it should return the shared journal id
      ✓ it should return the all the private journals id
    /POST/:id journal
      ✓ it should do nothing on invalid journal
      ✓ it should do nothing on inexisting journal
      ✓ it should add entry in the journal (1/2)
      ✓ it should add another entry in the journal (2/2)
    /GET/:id journal
      ✓ it should do nothing on invalid journal
      ✓ it should do nothing on inexisting journal
      ✓ it should return all entries on existing journal with no filter
      ✓ it should return all entries with fields[text,metadata]
      ✓ it should return all entries with even with inexisting fields (43ms)
      ✓ it should return nothing when filtered on inexisting activity id
      ✓ it should return nothing when filtered on current stime
      ✓ it should return all entries when not filtered on favorite
      ✓ it should return favorite entry when filtered on favorite
      ✓ it should return non favorite entry when filtered on not favorite
      ✓ it should return one entry when filtered on part of existing title
      ✓ it should return one entry when filtered on non existing title
      ✓ it should return filtered entries with fields[text,metadata] and filter on activity id
      ✓ it should return nothing when filtered on inexisting object id
      ✓ it should return filtered entries with fields[text,metadata] on object id
      ✓ it should return all entries when filtered on user id
      ✓ it should return all entries sorted on timestamp(desc)
    /PUT/:id journal
      ✓ it should update nothing on invalid journal id
      ✓ it should update nothing on invalid journal id
      ✓ it should update existing entry with valid journal id
      ✓ it should update an entry in journal with valid id
    /DELETE/:id journal
      ✓ it should delete nothing on invalid id
      ✓ it should delete nothing on inexisting id
      ✓ it should not delete the entry with valid journal id and invalid oid
      ✓ it should delete the entry with valid journal id and oid
      ✓ it should delete the journal with valid id

  Stats
    /POST stats
      ✓ it should insert stats
    /GET stats
      ✓ it should return all stats for the user
      ✓ it should return fields for the stats
    /DELETE stats
      ✓ it should delete all stats for the user

  Users
    /POST users
      ✓ it should add a student user
      ✓ it should not add duplicate student user
    /GET/:id users
      ✓ it should return nothing on invalid id
      ✓ it should return nothing on inexisting id
      ✓ it should return right user by id
    /GET users
      ✓ it should return all the users
      ✓ it should return all the fields for admin user
      ✓ it should return all the fields for student user
      ✓ it should return all matched users with partial search
      ✓ it should return all users after a timestamp
      ✓ it should not return users for a future timestamp
    /PUT/:id users
      ✓ it should do nothing on invalid user
      ✓ it should do nothing on inexisting user
      ✓ it should update the valid user
      ✓ it should not update the user with duplicate name
    /DELETE/:id users
      ✓ it should do nothing on invalid user
      ✓ it should not remove an inexisting user
      ✓ it should remove the valid user


  76 passing (3s)

**Comment by ashish0910** [comment]
@tarunsinghal92 Also the file_size of journal is set to 0 as mentioned in https://github.com/llaske/sugarizer-server/blob/e4c373896485b850435366a5dac64aac66f0fd37/api/controller/journal.js#L462
why is it so ?

**Comment by NikhilM98** [comment]
@ashish0910 Files Changed tab in the PR is showing changes in `dashboard/public/l10n/locales.ini`. I'm not sure why. Those changes are already merged in the `dev` branch.
Is is really necessary to sync `locales.ini` with dev? I think git will take care of it when your PR gets merged.

**Comment by ashish0910** [comment]
@NikhilM98 I had to sync it with dev because earlier it was showing some changes in that so I synced it with dev .

**Comment by NikhilM98** [comment]
@ashish0910 I'm also not sure about it.
Maybe you can try to reset your git head. Then reset files that are not required to be changed. Then add, commit and force push the code on your branch.
If there are conflicts then rebase your branch with the upstream dev.

P.S. If there are too many commits related to single issue, it as a good idea to squash the commits.

**Comment by ashish0910** [comment]
@NikhilM98 , at the end it shows Everything up-to-date . May be we can let it merge with dev branch as locales.ini file is in sync with dev branch . @tarunsinghal92

**Comment by llaske** [comment]
Using a new combobox to select the sorted column is too complex in my opinion.
My wish is to allow user to select the sorted column just by clicking on the column header.

**Comment by ashish0910** [comment]
@llaske sure will change that accordingly .

**Comment by ashish0910** [comment]
@llaske  changes Done in #77

**Pull request closed without merging**